enterprise/reporting/reporting.go

```diff
@@ -460,7 +493,9 @@ func (r *DefaultReporter) mainLoop(ctx context.Context, c types.SyncerConfig) {
 				if err != nil {
 					return err
 				}
-				_, err = dbHandle.Exec(`DELETE FROM `+ReportsTable+` WHERE reported_at = $1`, reportedAt)
+				// Use the same aggregationIntervalMin value that was used to query the reports in getReports()
+				bucketStart, bucketEnd := r.getAggregationBucketMinute(reportedAt, aggregationIntervalMin)
+				_, err = dbHandle.Exec(`DELETE FROM `+ReportsTable+` WHERE reported_at >= $1 and reported_at < $2`, bucketStart, bucketEnd)
```

Comment:
why are we deleting from bucket only we can delete all the entries before last reported time?

Reply:
we can delete all the reports `< bucketEnd`, I will change the filter in `getReports()` too.

Reply:
[minor]: Very unlikely, but what if there are rows added with older time while we are processing current window

Reply:
yeah, revert the changes, its better to add the bucketStart